public/app/features/geo/editor/locationModeEditor.tsx

```diff
@@ -0,0 +1,120 @@
+
+const MODE_OPTIONS = [
+  {
+    value: FrameGeometrySourceMode.Auto,
+    label: 'Auto',
+    ariaLabel: selectors.components.Transforms.SpatialOperations.location.autoOption,
+    description: 'Automatically identify location data based on default field names',
+  },
+  {
+    value: FrameGeometrySourceMode.Coords,
+    label: 'Coords',
+    ariaLabel: selectors.components.Transforms.SpatialOperations.location.coords.option,
+    description: 'Specify latitude and longitude fields',
+  },
+  {
+    value: FrameGeometrySourceMode.Geohash,
+    label: 'Geohash',
+    ariaLabel: selectors.components.Transforms.SpatialOperations.location.geohash.option,
+    description: 'Specify geohash field',
+  },
+  {
+    value: FrameGeometrySourceMode.Lookup,
+    label: 'Lookup',
+    ariaLabel: selectors.components.Transforms.SpatialOperations.location.lookup.option,
+    description: 'Specify Gazetteer and lookup field',
+  },
+];
+
+interface ModeEditorSettings {
+  data?: DataFrame[];
+  source?: FrameGeometrySource;
+}
+
+const helpUrl = 'https://grafana.com/docs/grafana/latest/panels-visualizations/visualizations/geomap/#location';
+
+export const LocationModeEditor = ({
+  value,
+  onChange,
+  context,
+  item,
+}: StandardEditorProps<string, ModeEditorSettings, unknown, unknown>) => {
+  const [info, setInfo] = useState<FrameGeometryField>();
+
+  useEffect(() => {
+    if (item.settings?.source && item.settings?.data?.length && item.settings.data[0]) {
+      getLocationMatchers(item.settings.source).then((location) => {
+        if (item.settings && item.settings.data) {
+          setInfo(getGeometryField(item.settings.data[0], location));
+        }
+      });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [item.settings]);
+
+  const styles = useStyles2(getStyles);
+
+  const dataValidation = () => {
+    if (info) {
+      if (info.warning) {
+        return (
+          <Alert
+            title={info.warning}
+            severity="info"
+            buttonContent={<Icon name="question-circle" size="xl" />}
+            onRemove={() => {
+              const newWindow = window.open(helpUrl, '_blank', 'noopener,noreferrer');
+              if (newWindow) {
+                newWindow.opener = null;
+              }
+            }}
+          />
+        );
+      } else if (value === FrameGeometrySourceMode.Auto && info.description) {
+        return <span className={styles.autoSuccess}>{info.description}</span>;
+      }
+    }
+    return null;
+  };
+
+  return (
+    <>
+      <Select
+        options={MODE_OPTIONS}
+        value={value}
+        onChange={(v) => {
+          onChange(v.value);
+        }}
+      />
+      <HorizontalGroup className={styles.hGroup}>{dataValidation()}</HorizontalGroup>
+    </>
+  );
+};
+
+const getStyles = (theme: GrafanaTheme2) => {
+  return {
+    autoSuccess: css`
+      color: ${theme.colors.success.text};
```

Comment:
Maybe we can change this to the normal font color (default)?
![Screenshot 2022-11-21 at 5 22 29 PM](https://user-images.githubusercontent.com/22381771/203184644-2bab9403-aa25-4cdf-a779-5d2ee72eb3d9.png)